### PR TITLE
add generic feature extraction function

### PR DESCRIPTION
**Pull request by christophmluscher**
Adds a function to the meta system class to extract generic features.

### CODE REVIEW

**Review by michelwi — APPROVED** [review]

File: meta/system.py [hunk 1]
```diff
@@ -287,6 +287,26 @@ def gt_features(self, corpus, prefix="", **kwargs):
...
+
+        feature_path = rasr.FlagDependentFlowAttribute(
+            "cache_mode",
+            {
+                "task_dependent": self.feature_caches[corpus][name],
+                "bundle": self.feature_bundles[corpus][name],
+            },
+        )
+        self.feature_flows[corpus][name] = features.basic_cache_flow(feature_path)
+        self.feature_flows[corpus][f"uncached_{name}"] = f.feature_dlow
```
Comment (JackTemaki) [inline meta/system.py:308]: `dlow` does not seem to be correct...

**Review by JackTemaki — CHANGES_REQUESTED** [review]

File: meta/system.py [hunk 2]
```diff
@@ -287,6 +287,26 @@ def gt_features(self, corpus, prefix="", **kwargs):
         self.feature_flows[corpus]["gt"] = features.basic_cache_flow(feature_path)
         self.feature_flows[corpus]["uncached_gt"] = f.feature_flow
 
+    def generic_features(
+        self, corpus, name, feature_flow, port_name_mapping, prefix="", **kwargs
+    ):
```
Comment (curufinwe) [inline meta/system.py:292]: docstring, especially for the relation between port_name_mapping and name

File: meta/system.py [hunk 3]
```diff
@@ -287,6 +287,26 @@ def gt_features(self, corpus, prefix="", **kwargs):
         self.feature_flows[corpus]["gt"] = features.basic_cache_flow(feature_path)
         self.feature_flows[corpus]["uncached_gt"] = f.feature_flow
 
+    def generic_features(
+        self, corpus, name, feature_flow, port_name_mapping, prefix="", **kwargs
```
Comment (curufinwe) [inline meta/system.py:291]: you could add a default port_name mapping, often the output of a flow network is "features"

**Review by curufinwe — CHANGES_REQUESTED** [review]

File: meta/system.py [hunk 4]
```diff
@@ -287,6 +287,41 @@ def gt_features(self, corpus, prefix="", **kwargs):
...
+        """
+        :param str corpus: corpus identifier
+        :param str name: feature identifier, like "mfcc". Also used in the naming of the output feature caches.
+        :param rasr.FlowNetwork feature_flow: definition of the RASR feature flow network
+        :param str port_name:
+                Maps the output of a flow network (key) to the name of the extracted feature (value).
+                name is the value and port_name is the key.
+                The port_name_mapping is used as parameter for the
+                FeaturesExtractionJob which normally has this structure:
+                {"features": name}
```
Comment (curufinwe) [inline meta/system.py:302]:
```suggestion
        :param str port_name: output port of the flow network to use
```

Comment (curufinwe) [reply meta/system.py:302]: Now that the user does not specify the port_name_mapping herself it's fine to not explain it anyomore.

**Review by curufinwe — CHANGES_REQUESTED** [review]

**Pull request merged**